loyalty_outlier_analysis.ipynb

```diff
@@ -0,0 +1,476 @@
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from sklearn.svm import SVC"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "train_and_validation_df.shape"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sampled_train_and_validation_df = train_and_validation_df.sample(frac=0.02, random_state=281316)\n",
+    "reduced_df = sampled_train_and_validation_df[sampled_train_and_validation_df.columns.difference(['first_active_month',\n",
+    "                                                                                              'target',\n",
+    "                                                                                              'authorized_flag_top',\n",
+    "                                                                                              'category_1_top',\n",
+    "                                                                                              'outlier'\n",
+    "                                                                                             ])]\n",
+    "reduced_df.fillna(reduced_df.mean(), inplace=True)\n",
+    "data_train, data_test, label_train, label_test = train_test_split(reduced_df,\n",
+    "                                                                  sampled_train_and_validation_df['outlier'],\n",
+    "                                                                  test_size=0.2)\n",
+    "data_train.shape"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "svc_clf = SVC(kernel='linear', class_weight='balanced', probability=True)\n",
+    "svc_clf.fit(data_train, label_train)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "evaluate_classification_results(svc_clf, data_train, label_train)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "evaluate_classification_results(svc_clf, data_test, label_test)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The problem with this is that, according to the documentation, \"The fit time complexity is more than quadratic with the number of samples\", meaning that it doesn't seem possible to use a large enough dataset. In any case, this method does not seem to do so well with precision, though recall is much better than with a decision tree."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Upsampling\n",
+    "\n",
+    "One more option is upsampling, that is, artificially inflating the number of outlier samples. Assuming there is any signal in the outlier data, this should enhance it so that it doesn't get lost in the noise."
```

Comment:
This is very interesting. I'd like to combine it with the outlier classifier I wrote ... there I was having the issue that it'd always predict everything as being not an outlier (giving 0% recall).